Mathlib/LinearAlgebra/FreeModule/StrongRankCondition.lean

```diff
@@ -19,46 +19,43 @@ module is well defined.
 
 ## Main result
 
+* `commRing_orzechProperty R` : `R` has the `OrzechProperty`.
 * `commRing_strongRankCondition R` : `R` has the `StrongRankCondition`.
 
 ## References
 
-We follow the proof given in https://mathoverflow.net/a/47846/7845.
+The original proof was adapt from <https://mathoverflow.net/a/47846/7845>.
 The argument is the following: it is enough to prove that for all `n`, there is no injective linear
 map `(Fin (n + 1) → R) →ₗ[R] Fin n → R`. Given such an `f`, we get by extension an injective
 linear map `g : (Fin (n + 1) → R) →ₗ[R] Fin (n + 1) → R`. Injectivity implies that `P`, the
 minimal polynomial of `g`, has non-zero constant term `a₀ ≠ 0`. But evaluating `0 = P(g)` at the
 vector `(0,...,0,1)` gives `a₀`, contradiction.
 
+The new proof dierctly proves that any commutative ring satisfies the `OrzechProperty`,
+see `Module.Finite.injective_of_surjective_of_submodule` and
+`Module.Finite.injective_of_surjective_of_injective`, which is a corollary of
+Noetherian rings satisfy the `OrzechProperty`
+(`IsNoetherianRing.orzechProperty`,
+see also `IsNoetherian.injective_of_surjective_of_submodule` and
+`IsNoetherian.injective_of_surjective_of_injective`)
+and Hilbert basis theorem.
+
```

Comment:
Add the references to the three papers we had seen, esp. Ribenboim.

Reply:
Added.